pandapower/convert_format.py

```diff
@@ -546,3 +550,39 @@ def _update_characteristics(net, elements_to_deserialize):
             continue
         c.interpolator_kind = "interp1d"
         c.kwargs = {"kind": c.__dict__.pop("kind"), "bounds_error": False, "fill_value": c.__dict__.pop("fill_value")}
+
+
+def convert_trafo_pst_logic(net):
+    """
+    Converts trafo and trafo3w phase shifter logic to version 3.0 or later
+    """
+    for trafotable in ["trafo", "trafo3w"]:
+        if trafotable in net and isinstance(net[trafotable], pd.DataFrame):
+            if net[trafotable].index.size > 0:
+                for t in ("", "2"):
+                    # drop old tap_phase_shifter flag
+                    if f"tap{t}_phase_shifter" in net[trafotable]:
+                        net[trafotable] = net[trafotable].drop(columns=f"tap{t}_phase_shifter")

```

Comment:
this is a bit brutal since tap{t}_phase_shifter = True means "Ideal Phase Shifter" and False means "Ratio". But since the information is redundant, the code below is correct IF the input is consistent.